libkode/printer.cpp

```diff
@@ -956,16 +951,47 @@ void Printer::printImplementation( const File &file, bool createHeaderInclude )
     filename.prepend( d->mOutputDirectory + '/' );
 
   QFile implementation( filename );
-  if ( !implementation.open( QIODevice::WriteOnly ) ) {
-    qWarning( "Can't open '%s' for writing.", qPrintable( filename ) );
-    return;
+  printCodeIntoFile(out, &implementation);
+}
+
+void Printer::printCodeIntoFile(const Code &code, QFile *file)
+{
+  QString outText = code.text();
+  bool identical = true;
+  QTextStream fileStream( file );
+  if (QFile::exists(file->fileName())) {
+    if ( !file->open( QIODevice::ReadOnly ) ) {
+      qWarning( "Can't open '%s' for reading.", qPrintable( file->fileName() ) );
+      return;
+    }
+    QTextStream codeStream( &outText );
+    QString fileLine, outLine;
+    while (fileStream.readLineInto(&fileLine) && codeStream.readLineInto(&outLine)) {
+      if (fileLine != outLine) {
+        identical = false;
+        break;
+      }
+    }
+    identical = fileStream.atEnd() && codeStream.atEnd();
```

Comment:
This overwrites the value of identical set inside the loop (making that "= false" statement useless).

If the *last* line differs, the current code will end up saying identical = true because of this condition.

I believe it should be `identical = identical && fileStream.atEnd() && codeStream.atEnd()`

Reply:
Fixed.